new_platforms/vscode-extension/assets/standaloneSolutionTemplateFolder/host/main.c

```diff
@@ -1,17 +1,16 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-#include <stdio.h>
-
 #include <openenclave/host.h>
+#include <stdio.h>
 
-int open_enclave();
+int open_enclave(int argc, const char* argv[]);
```

Comment:
Is there any reason why we prefer `open_enclave` and `close_enclave` instead of the create_enclave and terminate_enclave used in samples?

Reply:
Do you mean instead of `oe_create_enclave` and `oe_terminate_enclave`?

The idea is to share the enclave creation, call, and termination logic between the standalone and IoT Edge project types.

Notice that `host.c` lives under `baseSolutionTemplateFolder/host` while there are two `main.c` files: one under `standaloneSolutionTemplateFolder/host` and another under `edgeSolutionTemplateFolder/host`. The former has a simple `main` function, while the latter talks to Azure IoT Hub. However, both call into `host.c` to deal with the enclave.

During template expansion, the contents of `baseSolutionTemplateFolder` are always applied. Then, depending on the project type, the contents of either of `standaloneSolutionTemplateFolder` or `edgeSolutionTemplateFolder` are applied on top.

So this is all ultimately to reduce code duplication.

Reply:
Oh, I see, the more complex examples factor out this logic too into `create_enclave` and `terminate_enclave`, got it.

No reason for the disparity, I'll change the templates to have the same naming.

Reply:
Fixed.